app/controllers/job_templates_controller.rb

```diff
@@ -24,6 +24,26 @@ def preview
     end
   end
 
+  def import
+    contents = params[:imported_template][:template].respond_to?(:read) ? params[:imported_template][:template].read : nil
+
+    @template = JobTemplate.import(contents, :update => Foreman::Cast.to_bool(params[:imported_template][:overwrite]))
+    if @template && @template.save
+      flash[:notice] = _('Job template imported successfully.')
+      redirect_to job_templates_path(:search => "name = \"#{@template.name}\"")
+    else
+      @template ||= JobTemplate.import(contents, :build_new => true)
+      @template.valid?
+      flash[:warning] = _('Unable to save template. Correct highlighted errors')
+      render :action => 'new'
+    end
+  end
+
+  def export
+    find_resource unless @template.present?
```

Comment:
That's the API controller.  For some reason, the methods in the plain controller do find_resource in the first method call (see preview).  I assume because this one is inheriting from Foreman's templates controller and we didn't want to accidentally override any of their filters?